File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if not numbers: return []

        start = 0
        stop = len(numbers)-1

        while start < stop:
            currsum = numbers[start] + numbers[stop]

            if currsum == target:
                return [start + 1, stop +1]
            elif currsum > target:
                stop -= 1
            else:
                start += 1


        return []

File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
#include <unordered_map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map <int, int> dict;
        
        for (int i = 0; i < nums.size(); i ++){
            if (dict.contains(target-nums[i])){
                return {dict[target-nums[i]], i};
            } else {
                dict[nums[i]] = i; 
            }
        }
        return {};
    }
};
